### DISCUSSION TITLE
logo overlap on video streaming in ant medio android sdk

### DISCUSSION THREAD

**Discussion by yogeshkaharmis** [ant-media/Ant-Media-Server#6154] [General]
hello @mekya @USAMAWIZARD 
I want to show image on video streaming in ant medio android sdk. Please give me information for this.
how to showing image on top left Conner of the screen in video streaming using ant media android sdk. 
Please provide code and example to achieve this.
thank you

**Comment by USAMAWIZARD** [comment]
Hi @yogeshkaharmis 
you can easily archive this using filter plugin , [here](https://antmedia.io/discover-the-filter-plugin/) is the blog post for that . 
 https://antmedia.io/discover-the-filter-plugin/ please take a look into Movie Filter and Adding Logo section , please let me know if you face any problem

**Reply by yogeshkaharmis** [reply]
hello @USAMAWIZARD 
I am check the filter plugin.

I am getting many issue about work-
1 how to pass image path.
2 image is store in project assets folder so how to pass assets folder path.
3 how to get successful api response because all time getting fail response.

I am using below information.

api - http://test.antmedia.io:5080/LiveApp/rest/v2/filters/create 

parameter - {"filterId":"logodemo","inputStreams":["streamId123"],"outputStreams":["streamId123"],"videoFilter":"movie=/Users/Android/Downloads/logomis.png[logo];[in0][logo]overlay[out0]","audioFilter":"[in0]acopy[out0]","videoEnabled":"true","audioEnabled":"true","type":"asynchronous"}

response - {
    "success": false,
    "message": "Video filter graph can not be initiated:movie=/Users/Android/Downloads/logomis.png[logo];[in0][logo]overlay[out0]",
    "dataId": null,
    "errorId": 0
}

you can check above information and please tell me what was wrong.

**Reply by USAMAWIZARD** [reply]
@yogeshkaharmis can you please add the updated command here that fixed the issue after our meeting.

**Comment by yogeshkaharmis** [comment]
hello @USAMAWIZARD
issue fix and work is done.
thank you for help.

**Reply by USAMAWIZARD** [reply]
can you please share the way here so that it could help others

**Reply by yogeshkaharmis** [reply]
hello @USAMAWIZARD 
it is helpful for and I tell you how to doing this work.

api - https://stream.vuyu.online:5443/WebRTCApp/rest/v2/filters/create

In above api please change "stream.vuyu.online" is by your server url.

parameter - {
    "filterId": "logodemoP3lp2SetLYAIZNyL1709807729215",
    "inputStreams": [
        "P3lp2SetLYAIZNyL1709807729215"
    ],
    "outputStreams": [
        "P3lp2SetLYAIZNyL1709807729215"
    ],
    "videoFilter": "movie=/usr/local/antmedia/webapps/WebRTCApp/images/whitepng.png[logo];[in0][logo]overlay[out0]",
    "audioFilter": "[in0]acopy[out0]",
    "videoEnabled": "true",
    "audioEnabled": "true",
    "type": "synchronous"
}
in the above parameter 
1 filterId parameter in you can add your streamid for unique like -  logodemoP3lp2SetLYAIZNyL1709807729215 (P3lp2SetLYAIZNyL1709807729215 is streamId)
2 you can pass StreamId in the both parameter inputStreams , outputStreams tag.
3 videoFilter tag in your server image url add on place of "/usr/local/antmedia/webapps/WebRTCApp/images/whitepng.png"
4  type tag in pass "synchronous"
and remaining parameter value as same as above given.

Note - you first start streaming then call api then you can see the result is logo overlap in streaming.

thank you.
